### PR TITLE
Add 367 release notes

### PR DESCRIPTION
**Pull request by martint**
Closes https://github.com/trinodb/trino/issues/10301

### CODE REVIEW

File: docs/src/main/sphinx/release/release-367.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  the `hide-inaccessible-columns` configuration property is set to true. ({issue}`9991`)
+* Disable `SET AUTHORIZATION` when `VIEW` runs as `DEFINER`. ({issue}`10351`)
+
+## Web UI
+
+* Fix logout from Web UI when Oauth2 authentication is used. ({issue}`10299`)
+
+## Hive connector
+
+* Introduce metadata cache flush procedure. ({issue}`10251`)
```
Comment (ebyhr) [inline docs/src/main/sphinx/release/release-367.md:23]: How about:
```markdown
Add procedure `system.flush_metadata_cache()` to flush metadata caches.
```

**Review by kokosing — APPROVED** [review]

File: docs/src/main/sphinx/release/release-367.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Release 367 (XX Dec 2021)
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-367.md:1]: remainder

File: docs/src/main/sphinx/release/release-367.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## General
+
+* Improve retryable query performance by adding support for dynamic filtering. ({issue}`10274`)
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-367.md:5]:
```suggestion
* Improve performance when query retries are enabled by adding support for dynamic filtering. ({issue}`10274`)
```

File: docs/src/main/sphinx/release/release-367.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+* Improve `CREATE VIEW` error message when table or materialized view already exists. ({issue}`10186`)
+* Fix failure in `min_by` and `max_by` aggregation execution. ({issue}`10347`)
+* Fix planning failure when selecting all fields from a row type. ({issue}`10321`)
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-367.md:8]: IIUC the issue correctly it was not only regarding `ROW(..).*` but also if subscript on row was used in query directly `ROW(...)[1]`.

Then maybe such phrasing would be more accurate:
```
Fix failure of specific queries which accessed fields of `row` column using index based subscript operator (`ROW(...)[n]`) or selecting all row fields (`ROW(..).*`)
```

File: docs/src/main/sphinx/release/release-367.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+* Fix potential hang when processing broadcast joins. ({issue}`10344`)
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-367.md:9]:
```suggestion
* Fix bug when specific queries which use broadcast joins could hang and never complete. ({issue}`10344`)
```

File: docs/src/main/sphinx/release/release-367.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+* Fix failure when row or array in `VALUES` cause contains nulls. ({issue}`10141`)
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-367.md:10]: This does not read correctly.
@martint you have context. Can you fix phrasing.

File: docs/src/main/sphinx/release/release-367.md (hunk 4 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-367.md:8]: Yes, it is. I was trying to describe the general area that might be affected without making it too convoluted. I'll update with a variant based on your wording.

File: docs/src/main/sphinx/release/release-367.md (hunk 6 above)
Comment (martint) [reply docs/src/main/sphinx/release/release-367.md:10]: Yes, the word "cause" shouldn't be there.

Comment (martint) [reply docs/src/main/sphinx/release/release-367.md:10]: It should be "clause"

File: docs/src/main/sphinx/release/release-367.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+## Security
+
+* Hide inaccessible columns from `SELECT *` statement when  
+  the `hide-inaccessible-columns` configuration property is set to true. ({issue}`9991`)
+* Disable `SET AUTHORIZATION` when `VIEW` runs as `DEFINER`. ({issue}`10351`)
+
+## Web UI
+
+* Fix logout from Web UI when Oauth2 authentication is used. ({issue}`10299`)
```
Comment (losipiuk) [inline docs/src/main/sphinx/release/release-367.md:20]:
```suggestion
* Improve UX by introducing a new landing page for logout flow when Oauth2 authentication is used. ({issue}`10299`)
```

**Review by losipiuk — APPROVED** [review]
% comments

File: docs/src/main/sphinx/release/release-367.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Release 367 (22 Dec 2021)
+
+## General
+
+* Capture lineage details for columns from `WITH` clauses and subqueries. ({issue}`10272`)
+* Improve `CREATE VIEW` error message when table or materialized view already exists. ({issue}`10186`)
+* Disallow query retries when connectors cannot perform them safely ({issue}`10064`)
```
Comment (phd3) [inline docs/src/main/sphinx/release/release-367.md:7]: nit: `.`

**Review by phd3 — APPROVED** [review]

**Pull request merged**